### ISSUE TITLE
CVE-2024-24791 in v3.122.0

### ISSUE THREAD

**Issue by SivaneshLogandurai** [pulumi/pulumi#16571]
### What happened?

Our scanning jobs have identified a new CVE "[CVE-2024-24791](https://www.cve.org/CVERecord?id=CVE-2024-24791)" in the pulumi 3.122.0. This is an issue with the Go standard library net/http.

### Example

Scan result

```
{
      "Target": "home/sl/.pulumi/bin/pulumi",
      "Class": "lang-pkgs",
      "Type": "gobinary",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2024-24791",
          "PkgName": "stdlib",
          "PkgIdentifier": {
            "PURL": "pkg:golang/stdlib@1.22.4",
            "UID": "69981ae570556369"
          },
          "InstalledVersion": "1.22.4",
          "FixedVersion": "1.21.12, 1.22.5",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:63f8b2cbd7add8bc0c9f354eaae360e9e45c19bf41f6edb5d3d2a37ea55f27f0",
            "DiffID": "sha256:24d32a2a221c7ba4f1f3737aea499fc5bf8eac8f8391f8ec9a62334045eb11e1"
          },
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-24791",
          "DataSource": {
            "ID": "govulndb",
            "Name": "The Go Vulnerability Database",
            "URL": "https://pkg.go.dev/vuln/"
          },
          "Description": "The net/http HTTP/1.1 client mishandled the case where a server responds to a request with an \"Expect: 100-continue\" header with a non-informational (200 or higher) status. This mishandling could leave a client connection in an invalid state, where the next request sent on the connection will fail. An attacker sending a request to a net/http/httputil.ReverseProxy proxy can exploit this mishandling to cause a denial of service by sending \"Expect: 100-continue\" requests which elicit a non-informational response from the backend. Each such request leaves the proxy with an invalid connection, and causes one subsequent request using that connection to fail.",
          "Severity": "UNKNOWN",
          "References": [
            "https://go.dev/cl/591255",
            "https://go.dev/issue/67555",
            "https://groups.google.com/g/golang-dev/c/t0rK-qHBqzY/m/6MMoAZkMAgAJ",
            "https://pkg.go.dev/vuln/GO-2024-2963"
          ],
          "PublishedDate": "2024-07-02T22:15:04.833Z",
          "LastModifiedDate": "2024-07-02T22:15:04.833Z"
        }
      ]
    },
    {
      "Target": "home/sl/.pulumi/bin/pulumi-language-python",
      "Class": "lang-pkgs",
      "Type": "gobinary",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2024-24791",
          "PkgName": "stdlib",
          "PkgIdentifier": {
            "PURL": "pkg:golang/stdlib@1.22.4",
            "UID": "6bd865116ccc0efe"
          },
          "InstalledVersion": "1.22.4",
          "FixedVersion": "1.21.12, 1.22.5",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:63f8b2cbd7add8bc0c9f354eaae360e9e45c19bf41f6edb5d3d2a37ea55f27f0",
            "DiffID": "sha256:24d32a2a221c7ba4f1f3737aea499fc5bf8eac8f8391f8ec9a62334045eb11e1"
          },
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-24791",
          "DataSource": {
            "ID": "govulndb",
            "Name": "The Go Vulnerability Database",
            "URL": "https://pkg.go.dev/vuln/"
          },
          "Description": "The net/http HTTP/1.1 client mishandled the case where a server responds to a request with an \"Expect: 100-continue\" header with a non-informational (200 or higher) status. This mishandling could leave a client connection in an invalid state, where the next request sent on the connection will fail. An attacker sending a request to a net/http/httputil.ReverseProxy proxy can exploit this mishandling to cause a denial of service by sending \"Expect: 100-continue\" requests which elicit a non-informational response from the backend. Each such request leaves the proxy with an invalid connection, and causes one subsequent request using that connection to fail.",
          "Severity": "UNKNOWN",
          "References": [
            "https://go.dev/cl/591255",
            "https://go.dev/issue/67555",
            "https://groups.google.com/g/golang-dev/c/t0rK-qHBqzY/m/6MMoAZkMAgAJ",
            "https://pkg.go.dev/vuln/GO-2024-2963"
          ],
          "PublishedDate": "2024-07-02T22:15:04.833Z",
          "LastModifiedDate": "2024-07-02T22:15:04.833Z"
        }
      ]
    }
```

### Output of `pulumi about`

Using pulum v3.122.0

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
This will be addressed in the next release when we build with go1.22.5.

**Comment by justinvp** [comment]
This is now fixed with [v3.123.0](https://github.com/pulumi/pulumi/releases/tag/v3.123.0) (the release is currently rolling out).
